nixos/modules/services/misc/sr-ht/builds.nix

```diff
@@ -0,0 +1,202 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.srht;
+  bcfg = cfg.builds;
+in {
+  options = {
+    services.srht.builds = {
+      redis = mkOption {
+        type = types.str;
+        default = "redis://localhost:6379/0";
+        description = ''
+          The redis connection used for the Celery worker.
+        '';
+      };
+
+      # TODO: Assertion
+      oauth-client.id = mkOption {
+        type = types.either types.int types.str;
+        default = "CHANGEME";
+        description = ''
+          OAuth client ID for meta.sr.ht from builds.sr.ht, register your client at
+          meta.example.org/oauth.
+        '';
+      };
+
+      # TODO: Assertion
+      oauth-client.secret = mkOption {
+        type = types.str;
+        default = "CHANGEME";
+        description = ''
+          OAuth client secret for meta.sr.ht from builds.sr.ht, register your client at
+          meta.example.org/oauth.
+        '';
+      };
+
+      worker.user = mkOption {
```

Comment:
Does the user need to be a "real" user or would a systemd `DynamicUser` be fine?

Reply:
Just did a bit of reading and using `DynamicUser` might be possible if the logs go into `/var` or `/var/tmp` and if I can serve the directory through nginx, since I don't know if it's possible to access it with `PrivateTmp` on.

Reply:
Do note that `DynamicUser` is currently broken in `18.09`   (I fixed it in `master` and `18.09`)

Reply:
I don't there are any plans to backport this anyway, seeing how `19.03` is coming soon.